### PR TITLE
Add extension import in packageresolve.targets

### PR DESCRIPTION
**Pull request by alexperovich**
This is to support the source indexer.

/cc: @weshaggard @ericstj

### CODE REVIEW

**Comment by weshaggard** [comment]
What do you need to plug into this spot from the source indexer?

**Comment by alexperovich** [comment]
The targets that rewrite package references to live built binaries.

**Comment by weshaggard** [comment]
Why does it need an extension point here then? You should be able to include that target anywhere and just hook it into the correct target chain.

**Comment by alexperovich** [comment]
Well it needs at least one extension point. I put it here because it is logically a modification of packageresolve.targets.

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/packageresolve.targets [hunk 1]
```diff
@@ -183,4 +183,7 @@
     <Message Importance="Low"
              Text="Removed all ResolvedTagetingPackReferences that were not specified explicitly as a TargetingPackReference=[@(TargetingPackReference)]. PackageReferencesToRemove=[@(PackageReferencesToRemove)]." />
   </Target>
+  
+  <Import Project="$(CustomAfterPackageResolveTargets)" Condition="'$(CustomAfterPackageResolveTargets)' != '' AND Exists('$(CustomAfterPackageResolveTargets)')"/>
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/packageresolve.targets]: Lets make this a little more general and put it in build.common.targets.

Also the Exists check should be enough as you don't need to check for empty as well.

**Pull request merged**